### PR TITLE
Hide with inline

### LINKED ISSUE #553: Hide should have a default display of inline
Since `Hide` is often used to shorten sentences, its inherent `display: block` is often undesirable.  It should default to use `display: inline` instead.

### PR DESCRIPTION
**Pull request by donnobot**
Added inline prop for hide
Fixes #553

### CODE REVIEW

File: packages/core/test/Hide.js [hunk 1]
```diff
@@ -36,4 +36,10 @@ describe('Hide', () => {
     const json = rendererCreateWithTheme(<Hide print />).toJSON()
     expect(json).toMatchSnapshot()
   })
+
+  test('renders with inline prop', () => {
+    const json = rendererCreateWithTheme(<Hide inline />).toJSON()
+    expect(json).toMatchSnapshot()
+    expect(json).toHaveStyleRule('display', 'inline')
```
Comment (BeniCheni) [inline packages/core/test/Hide.js]: 👍

File: packages/core/src/Hide.js [hunk 2]
```diff
@@ -34,6 +34,7 @@ const Hide = styled(Box)`
   ${hidden('xs')} ${hidden('sm')} ${hidden('md')} ${hidden('lg')} ${hidden(
   'xl'
 )} ${hidden('print')};
+  display: ${props => (props.inline ? 'inline;' : null)}
```
Comment (sdalonzo) [inline packages/core/src/Hide.js]: no propType?

File: packages/core/storybook/Hide.js [hunk 3]
```diff
@@ -22,5 +22,8 @@ storiesOf('Hide', module).add('Hide', () => (
     <Hide print p={2} bg="yellow">
       Hide print
     </Hide>
+    <Hide inline p={2} bg="blue">
```
Comment (sdalonzo) [inline packages/core/storybook/Hide.js]: without a `sm`, `md`, etc, Hide is effectively a no-op

**Comment by sdalonzo** [comment]
This also doesn't satisfy the requirement of the issue https://github.com/pricelinelabs/design-system/issues/553: `Hide.text` should be declared as a variation of Hide, that uses a `span` (`<Text.span>`) instead of <div>, but otherwise functions the same

**Pull request closed without merging**